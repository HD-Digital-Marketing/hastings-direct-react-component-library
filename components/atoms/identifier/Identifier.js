import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './identifier.scss'

const Identifier = props => {

    const {
        isReg = false,
        isSmall = false,
        className = '',
        children
    } = props

    const identifierClass = classNames('a-identifier', className, {
        'a-identifier--reg': isReg,
        'a-identifier--small': isSmall,
    })

    return(
        <span className={identifierClass}>{children}</span>
    )
}

Identifier.propTypes = {
    isReg: PropTypes.bool,
    isSmall: PropTypes.bool,
    theme: PropTypes.string,
    className: PropTypes.string,
    children: PropTypes.string
}

export default Identifier
