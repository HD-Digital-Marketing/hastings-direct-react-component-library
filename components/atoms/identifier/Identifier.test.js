import React from 'react'
import renderer from 'react-test-renderer'
import Identifier from './Identifier'

describe('Check the identifier renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create(
                <Identifier isReg={true} isSmall={true}>SD13 FWW</Identifier>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()

    })
})