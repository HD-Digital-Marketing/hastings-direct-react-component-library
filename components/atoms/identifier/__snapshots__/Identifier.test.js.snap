// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the identifier renders it should match the snapshot 1`] = `
<span
  className="a-identifier a-identifier--reg a-identifier--small"
>
  SD13 FWW
</span>
`;
